cmd/egress-cni-plugin/egressContext.go

```diff
@@ -233,101 +239,426 @@ func (c *EgressContext) SetupHostVeth(vethName string) error {
+			//Retrieve IP addresses assigned to the interface/link
+			contIPAddrs, _err = ec.Link.AddrList(link, ipFamily)
+			if _err != nil {
+				return fmt.Errorf("failed to get IP addresses for %s: %v", ec.NetConf.IfName, _err)
 			}
 
 			//Delete the interface/link.
-			if err = c.Link.LinkDel(iface); err != nil {
-				return fmt.Errorf("failed to delete %q: %v", c.NetConf.IfName, err)
+			_err = ec.Link.LinkDel(link)
+			if _err != nil {
+				if !cniutils.IsLinkNotFoundError(_err) {
+					ec.Log.Debugf("failed to delete veth %s in container: %v", ec.NetConf.IfName, _err)
+				}
+			} else {
+				ec.Log.Debugf("Successfully deleted veth %s in container", ec.NetConf.IfName)
 			}
+			return _err
+		})
 
-			for _, addr := range addrs {
-				ipnets = append(ipnets, addr.IPNet)
+		//DEL should be best-effort. We should clean up as much as we can and avoid returning error
+		if err != nil {
+			ec.Log.Debugf("DEL: Executing in container ns errored out, returning", err)
+		}
+	}
+
+	err = ec.cleanupSnat(ipv4, contIPAddrs)
+	if err != nil {
+		ec.Log.Errorf("Delete host SNAT chain/rule for ip %s failed: %v.", contIPAddrs, err)
+	}
+
+	return nil
+}
+
+// cmdAddEgressV6 exec necessary settings to support IPv6 egress traffic in EKS IPv4 cluster
+func (ec *egressContext) cmdAddEgressV6() (err error) {
+	// Per best practice, a new veth pair is created between container ns and node ns
+	// this newly created veth pair is used for container's egress IPv6 traffic
+	// NOTE:
+	//	1. link-local IPv6 addresses are automatically assigned to veth's both ends.
+	//	2. unique-local IPv6 address allocated from host-local IPAM plugin is assigned to veth's container end only
+	//  3. veth node end has no unique-local IPv6 address assigned, only link-local IPv6 address
+	//  4. container IPv6 egress traffic go through node primary interface (eth0) which has an IPv6 global unicast address
+	//  5. IPv6 egress traffic of all containers in a node shares node primary interface (eth0) through SNAT
+
+	if ec.IPTablesIface == nil {
+		if ec.IPTablesIface, err = ec.IptCreator(iptables.ProtocolIPv6); err != nil {
+			ec.Log.Error("command ip6tables not found")
+			return err
+		}
+	}
+	// first disable IPv6 on container's primary interface (eth0)
+	err = ec.disableInterfaceIPv6(ec.ArgsIfName)
+	if err != nil {
+		ec.Log.Errorf("failed to disable IPv6 on container interface %s", ec.ArgsIfName)
+		return err
+	}
+
+	hostInterface, containerInterface, err := ec.setupContainerVethV6()
+	if err != nil {
+		ec.Log.Errorf("veth created failed, ns: %s name: %s, mtu: %d, ipam-result: %+v err: %v",
+			ec.NsPath, ec.NetConf.IfName, ec.Mtu, *ec.TmpResult, err)
+		return err
+	}
+	ec.Log.Debugf("veth pair created for container IPv6 egress traffic, container interface: %s ,host interface: %s",
+		containerInterface.Name, hostInterface.Name)
+
+	containerIPv6, err := ec.getContainerIpv6GlobalAddrs(containerInterface.Name)
```

Comment:
Sorry, I think I asked this before, but I can't find the conversation. This function finds the global IPv6 address assigned to the container veth intf. Isn't this just the address we got from host-ipam? i.e. shouldn't it be available in `ec.TmpResult`?